### PR TITLE
Update setting metadata information

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Updating to remove old info that just needed to go and also to update with 5.8 changes

Some detail here https://lab.civicrm.org/dev/core/issues/495

### CODE REVIEW

File: docs/framework/setting.md [hunk 1]
```diff
@@ -41,35 +42,37 @@ Each file consists of a php snippet which returns an array. Array keys are strin
...
-| `quick_form_type` | Widget type (admin form)| e.g YesNo, Element  |
-| `html_type` | Html type (admin form)| Used when `quick_form_type` is "Element". |
+| `name` | Name of this setting| This is the same as the array key. Required |
+| `type` | Data type| String, Integer, Boolean (Array is discouraged as current uses of it do not have full handling and String  + a serialize key addresses most uses) |
+| `title` | Title (admin form)| note: use ts() or E::ts() for extensions|
+| `description` | Description (admin form)| note: use ts() or E::ts() for extensions|
+|`serialize`|define how an array of values are stored|ie CRM_Core_DAO::SERIALIZE_JSON is recommended. Other constants exist on the CRM_Core_DAO class|
+| `default` | Default value| Value to use if setting is not set. |
+| `add` | Version added to CiviCRM| |
+| `html_type` | Html type (admin form)| This is the preferred way to describe the html type as it is not quick form specific. It will be used it present. Syntax is lower case. e.g 'select', 'radio', 'checkboxes', 'checkbox', 'text', 'textarea', 'entity_reference|
```
Comment (seamuslee001) [inline docs/framework/setting.md]: @eileenmcnaughton just thinking i note int he DAO metadata we have `<type>Text</type>` or `<type>SelectDate</type>` but here we are going lowercase maybe there is some difference

Comment (eileenmcnaughton) [reply docs/framework/setting.md]: @seamuslee001 Yeah @mattwire was the one who concluded which way to go on casing & TBH I didn't go done the whole rabbit hole - but I think the type in DAO maps to 'quick_form_type' here?

**Comment by eileenmcnaughton** [comment]
@MegaphoneJon do you have merge right here?

**Comment by MegaphoneJon** [comment]
Yes, I can merge on all docs. Today is a holiday in the US but I'm going to do a pass soon.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: docs/framework/setting.md [hunk 2]
```diff
@@ -41,35 +42,37 @@ Each file consists of a php snippet which returns an array. Array keys are strin
...
-| title | Title (admin form)| note: do not use ts() |
-| description | Description (admin form)| note: do not use ts()  |
-| `help_text` | Popup Help (admin form)| note: do not use ts() |
 | `validate_callback` | Function to call for checking when submitted (admin form)| e.g `CRM_Utils_Rule::url` |
 | `on_change` | Callback function when this setting is altered e.g when you enable a component or logging| |
 | `is_domain` | Domain setting| see `civicrm_setting` table |
 | `is_contact` | Contact setting| see `civicrm_setting` table |
-| prefetch | Legacy support - will store the setting in the $config object| We need to migrate away from this |
 
 Deprecated settings properties are :
```
Comment (MegaphoneJon) [inline docs/framework/setting.md]: Should be "Example" not "Example Notes".

File: docs/framework/setting.md [hunk 3]
```diff
@@ -126,24 +133,39 @@ It is desirable to make this api handling of domain id part of the api layer for
 
 ## Adding a new Setting to CiviCRM Core.
 
-1. Choose a group of settings to add the new setting to
-2. Add metadata to the appropriate `.settings.php` file about the new setting
-3. Choose an admin setting page to add the configuration option to.
-4. Add the name and group to the `$_settings` array at the top of the file. The rest happens automatically.
-5. If the template for that form is not set up to handle new items automatically, add the necessary markup.
-6. Your setting will not be active until you run `bin/regen.sh` - run it and commit the changes to git.
+1. Add metadata to the appropriate `.settings.php` file about the new setting
```
Comment (MegaphoneJon) [inline docs/framework/setting.md]: Should be `.setting.php`, NOT `.settings.php`  I'll also note this above.

File: docs/framework/setting.md [hunk 4]
```diff
@@ -25,12 +25,13 @@ The "Settings" system developed incrementally:
 - v4.2 – Added public Settings API. Included support for CRUD'ing data from multiple sources (`civicrm_domain` and `civicrm_setting`). Migrated more settings data from `civicrm_domain` to `civicrm_setting`.
 - v4.3 to v4.6 – Incrementally migrate more settings data.
 - v4.7 – Added `Civi::settings()` and refactored init system. Finished migrating data from `civicrm_domain` to `civicrm_setting`.
+- v5.8 - Added Generic CiviCRM settings form.
 
 ## Settings Definition
 
 Settings are defined in the /settings directory, in files ending with `.settings.php`
```
Comment (MegaphoneJon) [inline docs/framework/setting.md]: Should be `.setting.php`, NOT `.settings.php`.

**Comment by eileenmcnaughton** [comment]
@MegaphoneJon I've mostly address your points. I wasn't quite sure about the Example Notes thing  - I can't merge to this repo

**Pull request merged**